### ISSUE TITLE
Responsive vimeo player with oEmbed?

### ISSUE THREAD

**Issue by cinemamike** [wrav/oembed#67]
Hello! I'm trying to work out how I can get a responsive player with oEmbed. I've been able to create an oEmbed field in craftcms, and bring it into my template with a width and height parameter... but I can't quite figure out how to make the player fit within a div (flex container) and grow/shrink when the browser window is moved.

Here's the code in my template:

<div class="videobox">
    {% set videoCall = entry.vimeoEmbed.render({width: 900,
        height: 506,}) %}

    {% set vidEmbed = videoCall %}
    {{ vidEmbed }}
</div>

I've also tried making some css changes to vimeo's iframe (and video tag) but it only results in a small video player. I've also tried putting a percentage in the parameters (i.e. width: 100%) but that also doesn't work...

I appreciate any thoughts you might have! Thanks!

**Comment by iparr** [comment]
This isn't really an _oembed_ problem, but try to use CSS to set the width to 100% of a container. This will override whatever you set your embed to use.

Then you can ensure that container is always sized correctly - again using CSS. I always use 16:9 for videos and the responsive ratio trick mentioned here: https://engageinteractive.co.uk/blog/top-10-scss-mixins

**Comment by reganlawton** [comment]
@iparr Thanks again for helping out other users. Yes CSS is the right solution here as not every provider that uses the `oembed` protocol supports W/H sizing params

**Comment by cinemamike** [comment]
I just wanted to say thanks for the tips! It turns out the problem I was having is that I had too many nested divs with too many styles (flexbox, position: relative, etc...) going on. Once I moved the video div out from under a few other divs my styles started working like normal.

I found this which helped a lot with my css styling: [https://stackoverflow.com/questions/49765190/how-to-embed-vimeo-video-with-responsive-design](url)

Thanks again!
